### ISSUE TITLE
Icon: components with icons are not aligned with ones that do not

### ISSUE THREAD

**Issue by benelan** [Esri/calcite-components#3430]
### Actual Behavior
The Avatars in the documentation are not aligned properly

### Expected Behavior
All of the avatars are aligned properly, regardless of the content

### Reproduction Steps and Sample

_Sample_: https://developers.arcgis.com/calcite-design-system/components/avatar/#sample

1. Reproducible in the "Variety of Avatars" documentation sample

### Relevant Info
from holistic feedback

_Reproducible version_: `@esri/calcite-components@1.0.0-beta.68`

- [x] CDN
- [ ] NPM package

Regression? _Last working version_: `@esri/calcite-components@1.0.0-<version>`

**Comment by macandcheese** [comment]
This occurs when using the `calcite-icon` component inside chip as well - probably independent of `avatar` : https://jsbin.com/miyowogisa/1/edit?html,output

**Comment by driskull** [comment]
Seems like some flex alignment is needed here.

**Comment by Elijbet** [comment]
@benelan Do you think you could put together a codepen for this outside of the documentation?

**Comment by benelan** [comment]
Thanks for fixing the avatar issue! However, I am still able to repro this issue in [Adam's example on `next`](https://codepen.io/benesri/pen/GRMMeYo). Sorry for moving the goal post, but I'd like to continue using this issue in case there is a fix that works for all affected components (maybe in `calcite-icon`?). I edited the title and description of the issue.

**Comment by Elijbet** [comment]
> Thanks for fixing the avatar issue! However, I am still able to repro this issue in [Adam's example on `next`](https://codepen.io/benesri/pen/GRMMeYo). Sorry for moving the goal post, but I'd like to continue using this issue in case there is a fix that works for all affected components (maybe in `calcite-icon`?). I edited the title and description of the issue.

Sorry I missed the other affected coponents. I'll get back to that in a bit.

**Comment by Elijbet** [comment]
@benelan: I’ve discussed this with @asangma and he thinks it might not be something that can be fixed from within our components. Seems like that's just what DIV does with images and SVGs. It just likes to add a few pixels to the bottom. He thinks we're up against default browser behavior in this case.

He is suggesting to design and build a ChipGroup (or Chips or ChipList) component that helps the layout and spacing between the chips. Like we do for PickList and PickListItem, etc.

What do you think @macandcheese?

**Comment by driskull** [comment]
If I add `vertical-align: top;` to the first chip it seems to fix it.

I also noticed that the icon is 16x16 but the text has a line-height of 19.

**Comment by macandcheese** [comment]
I'm not sure we need a wrapping container for individual chips ... we do have https://github.com/Esri/calcite-components/issues/1933 - which would have a parent group but to handle selection logic, but individual chips shouldn't be required to have a parent.

**Comment by asangma** [comment]
TLDR: I think that the alignment issue isn't a Calcite issue but resulting from default browser behavior.

> individual chips shouldn't be required to have a parent.

I agree.
I think that the misalignment is coming from the default behavior of the containing element, e.g. a `div` whose default `display` is `block`.

A simple fix is to add `display: flex` to the parent `div` or follow @driskull's suggestion, but, _and seriously BUT_, doing so seems like it'd be outside of our components and something that should be handled at the app level.

The way to do something like that in the scope of Calcite would be to introduce something like a "ChipGroup" or whatever. It would have appropriate styles applied for alignment, spacing between chips, spacing between groups, etc. But...I'd say it's outside the scope of this issue.

**Comment by macandcheese** [comment]
Yeah I guess the question is then.. assuming we add the above “chip filter group” - should that component handle alignment of “normal” non selectable chips? I don’t have a preference.

**Comment by asangma** [comment]
@macandcheese yeah...I think any chip in the chip group would get that alignment.

**Comment by benelan** [comment]
verified for avatar, a new issue has been created for a grouping component
